src/config.c

```diff
@@ -2166,10 +2166,6 @@ static int isValidAOFdirname(char *val, const char **err) {
         *err = "appenddirname can't be empty";
         return 0;
     }
-    if (includeSpace(val)) {
```

Comment:
Why remove this restriction? Do you think it is best practice to include spaces in dirname? This is a new configuration, let's put an end to this ugly design in the first place.

Reply:
when first reading this, i tend to agree.
but when thinking of it further, i'm not actually sure what's printable and what's not.
some people (certainly not me), may wanna use Hebrew or Chinese in their file names?

Reply:
@chenyang8094 I think it's not good practice, agree with your examples and if I had to build this from scratch I'd prevent that. That said, I see several reasons to avoid this limitation:

* Consistency: it is confusing that some parameters apply filename restrictions that others don't.
* User experience: as a user, if I had some reason (or just bad style) to use weird `appendfilename`, it may be painful for me not to be able to do that in `appenddirname` (even if technically it's not a regression).
* Low cost: the cost of supporting that is very low, thanks to the fact that everything is already binary safe. Even the change to the handling of the manifest file was minimal and consistent with how `redis.conf` is handled.